internal/tiltfile/k8s.go

```diff
@@ -67,9 +83,23 @@ func (r *k8sResource) addRefSelector(selector container.RefSelector) {
 	r.refSelectors = append(r.refSelectors, selector)
 }
 
+func (r *k8sResource) addWorkloadEntity(workload k8s.K8sEntity, imageJSONPaths func(e k8s.K8sEntity) []k8s.JSONPath) error {
+	if r.hasWorkloadEntity {
+		return fmt.Errorf("tried to add workload %q to resource %q, when it already has workload %q", workload.Name(), r.name, r.workloadEntity.Name())
```

Comment:
can you add a test case for this error? i don't totally understand what problem this is trying to prevent.